### PR TITLE
Passing external pointer to static calls

### PR DESCRIPTION
**Pull request by PavelBal**

### CODE REVIEW

**Comment by PavelBal** [comment]
@Ian-Du Please solve the conflicts.

Also the commit was saying "in-progress", is it ready now?

File: src/rSharp.cpp [hunk 1]
```diff
@@ -861,28 +859,34 @@ RSHARP_BOOL r_has_class(SEXP s, const char* classname) {
 	return FALSE_BOOL;
 }
 
-RSharpGenericValue* get_RSharp_generic_value(SEXP clrObj) {
+RSharpGenericValue* get_RSharp_generic_value_from_EXTPTR(SEXP clrObj)
+{
+	UNPROTECT(1);
```
Comment (PavelBal) [inline src/rSharp.cpp:864]: Why `UNPROTECT` here?

File: src/rSharp.cpp [hunk 2]
```diff
@@ -590,8 +590,6 @@ const char* get_type_full_name(RSharpGenericValue** genericValue) {
 	return (char*)(hr);
 }
 
-RSharpGenericValue* get_RSharp_generic_value(SEXP clrObj);
```
Comment (PavelBal) [inline src/rSharp.cpp:593]: Why removed the declaration but not added for the two new methods`?

File: src/rSharp.cpp [hunk 3]
```diff
@@ -861,28 +859,34 @@ RSHARP_BOOL r_has_class(SEXP s, const char* classname) {
...
 		PROTECT(clrobjSlotName = NEW_CHARACTER(1));
 		SET_STRING_ELT(clrobjSlotName, 0, mkChar("clrobj"));
 		a = getAttrib(clrObj, clrobjSlotName);
-		UNPROTECT(1);
-		if (a != NULL && a != R_NilValue && TYPEOF(a) == EXTPTRSXP) {
-			return GET_RSHARP_GENERIC_VALUE_FROM_EXTPTR(a);
-		}
-		else
-			return NULL;
+
```
Comment (PavelBal) [inline src/rSharp.cpp:881]: `UNROTECT` should be here and not in the other method, no?

**Comment by rwmcintosh** [comment]
Didn't we abandon this investigation?

**Comment by PavelBal** [comment]
No, as I had enough time while waiting for setting the library paths, I made this work ;) It is th R6 branch

**Pull request closed without merging**